Services/Calendar/classes/class.ilCalendarSchedule.php

```diff
@@ -594,15 +597,44 @@ protected function initPeriod(ilDate $seed)
 				}
 				else
 				{
-					//todo: previous implementation still taking more days than represented in the view.
 					$year_month = $seed->get(IL_CAL_FKT_DATE,'Y-m','UTC');
 					list($year,$month) = explode('-',$year_month);
 
+					#21716
 					$this->start = new ilDate($year_month.'-01',IL_CAL_DATE);
-					$this->start->increment(IL_CAL_DAY,-6);
 
+					$start_unix_time = $this->start->getUnixTime();
+
+					$start_day_of_week = (int)date('w', $start_unix_time);
+
+					$number_days_previous_month = $start_day_of_week;
+
+					if($this->weekstart === ilCalendarSettings::WEEK_START_MONDAY) {
+						$number_days_previous_month = $start_day_of_week - 1;
+					}
+
+					$this->start->increment(IL_CAL_DAY, -$number_days_previous_month);
```

Comment:
In the case first of month  is sunday:

seed->increment(IL_CAL_DAY, - - 1) => which means seed is incremented to 2. of month.

Reply:
You are right. My last commit fixes this issue.